src/node_api.cc

```diff
@@ -309,8 +311,12 @@ class ThreadSafeFunction : public node::AsyncResource {
                                "ERR_NAPI_TSFN_STOP_IDLE_LOOP",
                                "Failed to stop the idle loop") == napi_ok);
       } else {
-        v8::Local<v8::Function> js_cb =
-            v8::Local<v8::Function>::New(env->isolate, ref);
+        v8::Local<v8::Value> js_cb;
+        if (ref.IsEmpty()) {
+          js_cb = v8::Undefined(env->isolate);
+        } else {
+          js_cb = v8::Local<v8::Function>::New(env->isolate, ref);
+        }
         call_js_cb(env,
                    v8impl::JsValueFromV8LocalValue(js_cb),
```

Comment:
This then needs to be `js_cb.IsEmpty() ? nullptr : v8impl::JsValueFromV8LocalValue(js_cb)`

Reply:
This still needs changing.

Reply:
I mean this still needs `js_cb.IsEmpty() ? nullptr : v8impl::JsValueFromV8LocalValue(js_cb)`.